commit 00983b5b1d3c5451f8941d4c7258e8a0037a6738 (refs/changes/51/951/2)
Author: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date:   2016-11-28T15:16:06+00:00 (2 years, 10 months ago)
    
    PORTAL-2440 Add application key for metadata update in case it does not exist
    Reviewed By: Pedro P. Candel <pedro@joyent.com>

diff --git a/lib/index.js b/lib/index.js
index b4cfd2d..7904ddd 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -2292,6 +2292,9 @@ function modifyMetadata(user, key, md, account, cb) {
 
         self.getMetadata(user, key, account, function (err) {
             if (err && err.statusCode === 404) {
+                if (!md.cn) {
+                    md.cn = key;
+                }
                 self.add(dn, md, function (err2) {
                     if (err2) {
                         cb(translateError(err2));
